src/App.tsx

```diff
@@ -1,24 +1,215 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { FormEvent, useEffect, useState } from 'react';
+import { Todo } from './Components/Todo';
+import { Todo as TodoType } from './types/Todo';
+import {
+  getTodos,
+  addTodo,
+  removeTodo,
+  updateTodo,
+} from './api/todos';
+import { Notification } from './Components/Notification';
+import { Filter } from './Components/Filter';
 
-const USER_ID = 0;
+const USER_ID = 10539;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<TodoType[]>([]);
+  const [error, setError] = useState('');
+  const [filteredTodos, setFilteredTodos] = useState<TodoType[]>([]);
+  const [filter, setFilter] = useState('');
+  const [addValue, setAddValue] = useState('');
+  const [isInputDisable, setIsInputDisable] = useState(false);
+  const [tempTodo, setTempTodo] = useState<TodoType | null>(null);
+  const [deleteIds, setDeleteIds] = useState<number[]>([]);
+  const [isUpdating, setIsUpdating] = useState(false);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(response => {
+        setTodos(response);
+      })
+      .catch(() => setError('Unable to Load todos'));
+  }, [deleteIds, isUpdating]);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(response => {
+        setFilteredTodos(response);
+      })
+      .catch(() => setError('Unable to Load todos'));
+  }, [todos, tempTodo]);
+
+  useEffect(() => {
+    setTimeout(() => {
+      setError('');
+    }, 3000);
+  }, [error]);
+
+  useEffect(() => {
+    switch (filter) {
+      case 'Active': {
+        setFilteredTodos(todos.filter(todo => todo.completed === false));
+        break;
+      }
+
+      case 'Completed': {
+        setFilteredTodos(todos.filter(todo => todo.completed === true));
+        break;
+      }
+
+      default: {
+        setFilteredTodos(todos);
+      }
+    }
+  }, [filter]);
+
+  const handleFilter = (value: string) => {
+    setFilter(value);
+  };
+
+  const handleClose = () => {
+    setError('');
+  };
+
+  const handleAddTodo = (event: FormEvent) => {
+    event.preventDefault();
+
+    if (!addValue) {
+      setError('Title cant be empty');
+    } else {
+      const newTodo: TodoType = {
+        id: 0,
+        userId: USER_ID,
+        title: addValue,
+        completed: false,
+      };
+
+      setTempTodo(newTodo);
+      setIsInputDisable(true);
+
+      addTodo(USER_ID, newTodo).then(() => {
+        setAddValue('');
+        setIsInputDisable(false);
+        setTempTodo(null);
+      }).catch(() => setError('Unable to add a todo'));
+    }
+  };
+
+  const handleRemoveTodo = (id: number) => {
+    removeTodo(id).then(() => {
+      setTodos(todos.filter((todo) => todo.id !== id));
+    }).catch(() => setError('Unable to delete a todo'));
+  };
+
+  const handleComplete = (id: number, completed: boolean) => {
```

Comment:
The function updateTodo is being called with the wrong parameters. It should be called with an updated todo object instead of just the id and completed status.